### PR TITLE
Refactor redux timetables

### PR DESCRIPTION
**Pull request by JiaqiWang18**
### Description
Refactor timetable reducer

### CODE REVIEW

File: static/js/redux/constants/commonTypes.ts [hunk 1]
```diff
@@ -1,12 +1,113 @@
...
+  date_end: string;
+  time_start: string;
+  time_end: string;
+  location: string;
+  is_short_course: boolean;
+  section: number;
+}
+
+/**
+ * Secion contains multiple offerings
```
Comment (JJamesWWang) [inline static/js/redux/constants/commonTypes.ts:28]: typo here

File: static/js/redux/constants/commonTypes.ts [hunk 2]
```diff
@@ -1,12 +1,113 @@
...
-    title: String;
-    count: Number;
-    reacted: boolean;
+  title: string;
+  count: number;
+  reacted: boolean;
+}
+
+
+export interface Course{
```
Comment (JJamesWWang) [inline static/js/redux/constants/commonTypes.ts:90]: Space after Course?

File: static/js/redux/state/saving_timetable_reducer.jsx [hunk 3]
```diff
@@ -48,7 +48,7 @@ const savingTimetable = (state = initState, action) => {
     case ActionTypes.ADD_CUSTOM_SLOT:
     case ActionTypes.UPDATE_CUSTOM_SLOT:
     case ActionTypes.REMOVE_CUSTOM_SLOT:
-    case ActionTypes.CHANGE_ACTIVE_TIMETABLE:
+    case 'global/changeActiveTimetable':
```
Comment (JJamesWWang) [inline static/js/redux/state/saving_timetable_reducer.jsx:51]: Just curious, is there a reason why these are `'global/...'` strings and aren't directly referenced by importing the action?

**Review by JJamesWWang — APPROVED** [review]
LGTM

File: static/js/redux/state/saving_timetable_reducer.jsx (hunk 3 above)
Comment (JiaqiWang18) [reply static/js/redux/state/saving_timetable_reducer.jsx:51]: I think we can't directly use redux toolkit actions in traditional non-rtk reducers, so I think I have to use that name of the action which is the string. I will change it back to importing the action when I refactor this reducer

Comment (JJamesWWang) [reply static/js/redux/state/saving_timetable_reducer.jsx:51]: I see, that makes sense

File: static/js/redux/constants/commonTypes.ts (hunk 1 above)
Comment (JiaqiWang18) [reply static/js/redux/constants/commonTypes.ts:28]: I will fix those on the next pr

**Pull request merged**